zebra-chain/src/amount.rs

```diff
@@ -191,6 +193,94 @@ where
     }
 }
 
+impl<C> Hash for Amount<C>
+where
+    C: Constraint,
+{
+    /// Amounts with the same value are equal, even if they have different constraints
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.hash(state);
+    }
+}
```

Comment:
No reason to constrain this impl

```suggestion
impl<C> Hash for Amount<C> {
    /// Amounts with the same value are equal, even if they have different constraints
    fn hash<H: Hasher>(&self, state: &mut H) {
        self.0.hash(state);
    }
}
```